packages/components/src/Interactive.ts

```diff
@@ -0,0 +1,134 @@
+import _ from 'lodash/fp';
+import { createElement, forwardRef, ReactNode, useState } from 'react';
+
+import * as Utils from './utils';
+
+const allowedHoverVariables = [
+  'background',
+  'backgroundColor',
+  'border',
+  'color',
+  'boxShadow',
+  'opacity',
+  'textDecoration',
+];
+const pointerTags = ['button', 'area', 'a', 'select'];
+const pointerTypes = ['radio', 'checkbox', 'submit', 'button'];
+
+export type InteractiveProps = {
+  as: keyof JSX.IntrinsicElements;
+  children?: ReactNode;
+  className?: string;
+  disabled?: boolean;
+  hover?: React.CSSProperties;
+  role?: string;
+  style?: React.CSSProperties;
+  tabIndex?: number;
+  type?: string;
+  onBlur?: (event: any) => void;
+  onClick?: (event: any) => void;
+  onKeyDown?: (event: any) => void;
+  onMouseDown?: (event: any) => void;
+};
+
+export const Interactive: React.ForwardRefExoticComponent<InteractiveProps> = forwardRef(
+  (
+    {
+      as: TagName = 'div',
```

Comment:
Update to use the same API as AOU.

Reply:
Done in https://github.com/DataBiosphere/terra-ui/pull/3992/commits/77ef903895f2b1567182bec0756fcedb57bd3e8d.